components/server/src/workspace/gitpod-server-impl.ts

```diff
@@ -1713,6 +1702,35 @@ export class GitpodServerImpl implements GitpodServerWithTracing, Disposable {
         }
     }
 
+    public async setProjectConfiguration(ctx: TraceContext, projectId: string, configString: string): Promise<void> {
+        traceAPIParams(ctx, { projectId }); // filter configString because of size
+
+        const user = this.checkAndBlockUser("setProjectConfiguration");
+        await this.guardProjectOperation(user, projectId, "update");
+        const parseResult = this.gitpodParser.parse(configString);
+        if (parseResult.validationErrors) {
+            throw new Error(`This configuration could not be parsed: ${parseResult.validationErrors.join(', ')}`);
+        }
+        await this.projectsService.updateProjectPartial({
+            id: projectId,
+            config: { '.gitpod.yml': configString },
+        });
+    }
+
+    public async updateProjectPartial(ctx: TraceContext, partialProject: PartialProject): Promise<void> {
+        const user = this.checkUser("updateProjectPartial");
+        await this.guardProjectOperation(user, partialProject.id, "update");
+
+        const partial: PartialProject = { id: partialProject.id };
+        const allowedFields: (keyof Project)[] = ['settings']; // Don't add 'config' here! Please use `setProjectConfiguration` instead to parse & validate configs
```

Comment:
:100: Thx! IMO we should far often be more careful like this, especially when adding super-powerful methods that might change over time. :slightly_smiling_face: